src/Proxy.php

```diff
@@ -276,9 +279,16 @@ public function executeCallback(callable $callback, ...$arguments): void
      */
     private function fetchObject(): ?object
     {
-        $id = $this->objectManager()->getClassMetadata($this->class)->getIdentifierValues($this->object);
+        $objectManager = $this->objectManager();
 
-        return empty($id) ? null : $this->objectManager()->find($this->class, $id);
+        if ($objectManager instanceof DocumentManager) {
+            // mongo cannot have multi fields identifiers
+            $id = $objectManager->getClassMetadata($this->class)->getIdentifierValue($this->object);
```

Comment:
```suggestion        
      $classMetadata = $objectManager->getClassMetadata($this->class);
      if (!$classMetadata->isEmbeddedDocument) {
           $id = $classMetadata->getIdentifierValue($this->object);
      }
```

I tested the branch and without this change, creating fixtures fails for me when documents contain embeds. 
<img width="1252" alt="Screenshot 2022-01-04 at 9 58 55" src="https://user-images.githubusercontent.com/2118173/148034768-00e03a6d-8b12-49c1-9c7a-981090924fc8.png">
<img width="721" alt="Screenshot 2022-01-04 at 9 59 03" src="https://user-images.githubusercontent.com/2118173/148034781-c816b705-176b-4bdf-a3ec-c82eb35656e5.png">
<img width="1159" alt="Screenshot 2022-01-04 at 10 01 21" src="https://user-images.githubusercontent.com/2118173/148034782-6f6d57f7-1bb6-4797-83f1-409c32eef5e7.png">

Reply:
thanks @majermi4 you're right, we should handle case when dealing with embedded objects here

Reply:
here is a failing test case + fix

https://github.com/zenstruck/foundry/pull/240

Reply:
This fix has been merged into this PR/branch.